msal_extensions/token_cache.py

```diff
@@ -1,156 +1,113 @@
 
     def modify(self, credential_type, old_entry, new_key_value_pairs=None):
         with CrossPlatLock(self._lock_location):
-            if self._needs_refresh():
-                try:
-                    self.deserialize(self._read())
-                except IOError as exp:
-                    if exp.errno != errno.ENOENT:
-                        raise
-            super(FileTokenCache, self).modify(
+            self._reload_if_necessary()
+            super(PersistedTokenCache, self).modify(
                 credential_type,
                 old_entry,
                 new_key_value_pairs=new_key_value_pairs)
-            self._write(self.serialize())
-            self._last_sync = os.path.getmtime(self._cache_location)
+            self._persistence.save(self.serialize())
+            self._last_sync = time.time()
 
     def find(self, credential_type, **kwargs):  # pylint: disable=arguments-differ
         with CrossPlatLock(self._lock_location):
-            if self._needs_refresh():
-                try:
-                    self.deserialize(self._read())
-                except IOError as exp:
-                    if exp.errno != errno.ENOENT:
-                        raise
-                self._last_sync = time.time()
-            return super(FileTokenCache, self).find(credential_type, **kwargs)
-
-
-class UnencryptedTokenCache(FileTokenCache):
-    """An unprotected token cache to default to when no-platform specific option is available."""
-    def __init__(self, cache_location, **kwargs):
-        warnings.warn("You are using an unprotected token cache, "
-                      "because an encrypted option is not available for {}".format(sys.platform),
-                      RuntimeWarning)
-        super(UnencryptedTokenCache, self).__init__(cache_location, **kwargs)
-
+            self._reload_if_necessary()
+            return super(PersistedTokenCache, self).find(credential_type, **kwargs)
 
-class WindowsTokenCache(FileTokenCache):
-    """A SerializableTokenCache implementation which uses Win32 encryption APIs to protect your
-    tokens.
-    """
-    def __init__(self, cache_location, entropy='', **kwargs):
-        super(WindowsTokenCache, self).__init__(cache_location, **kwargs)
-        self._dp_agent = WindowsDataProtectionAgent(entropy=entropy)
-
-    def _write(self, contents):
-        with open(self._cache_location, 'wb') as handle:
-            handle.write(self._dp_agent.protect(contents))
-
-    def _read(self):
-        with open(self._cache_location, 'rb') as handle:
-            cipher_text = handle.read()
-        return self._dp_agent.unprotect(cipher_text)
 
+def build_token_cache(  # pylint: disable=too-many-arguments
+        location,
+        entropy='',  # For Windows
+        service_name=None, account_name=None,  # For OSX
+        schema_name="", attributes=None, fallback_to_plaintext=False,  # For Linux
+        **kwargs):  # Extra parameters will be relayed to LibsecretPersistence
```

Comment:
As a user, I'd prefer my own code to this helper because the union of options is too mysterious for me. It looks like sorting out how each persistence implementation uses which options is a prerequisite for calling this method anyway.

Reply:
@chlowell was eloquent during our offline discussion:
> I feel the helper doesn't really help me because it doesn't hide the particulars of persistence. To use it, I must learn which options apply on which platforms, and how; which ones are truly optional. Then I either pass all options at once or call the helper conditionally by platform. I prefer to write the conditional code because it would be easier for future generations to understand. At that point I may as well just import and configure persistence classes directly, that's nicely explicit.

The two latest commits demoted the `build_persistence()` and `build_token_cache()` into their own sample, i.e. removed them from API surface.

@chlowell please review again. If you like it, please approve. :-)